### DISCUSSION TITLE
cycle between multiple commands with the same shortcut

### DISCUSSION THREAD

**Discussion by dealfonso** [rxhanson/Rectangle#515] [Ideas]
I think that it would be interesting to enable the usage of the same shortcut for multiple commands, and cycle between them.

Use case:

I want to use Ctrl+Option+Up to put the windows in the upper half of the window, but if I press it again, it will use one quarter in the upper half of the window and once again will use the other quarter in the upper half; then a third touch will get to the whole upper half of the window.

I would assign the same command to all these commands, and rectangle would cycle between them. Perhaps in the same order that they were assigned. To avoid detecting where the window is (maybe it is already detected or not), a deadline for touching again (e.g. 2 secs.) could be implemented.

**Comment by rxhanson** [comment]
Thanks for the idea. I don't have plans to add this capability into Rectangle, but it is in the custom shortcut builder in my paid superset of Rectangle, [Rectangle Pro](https://rectangleapp.com/pro) (formerly Hookshot). Here's a post I wrote about it: https://medium.com/ryan-hanson/custom-window-dimension-shortcuts-in-macos-e7df347dfd24. Rectangle is intentionally really simple, which is why this capability is only in Hookshot.

**Comment by foebu** [comment]
Hi, is there an option to make all (or almost all) shortcuts cycle, or you need to create a new one for each shortcut we want to cycle over?

**Reply by rxhanson** [reply]
(Assuming you're referring to Rectangle Pro). There is not an option to make all shortcuts cycle. What exactly is your desired functionality, as in what size/positions are you trying to cycle through?

**Reply by foebu** [reply]
Yes, I was referring to RPRo. Anyway, I think that this configuration will do, and in case I just need to add the same for the up and down options:
![Screenshot 2022-11-16 at 09 44 12](https://user-images.githubusercontent.com/3037789/202146092-ca84fb47-7703-40a4-86cf-e1ec6efe11ba.png)
